docs/userguide/searchspace.md

```diff
@@ -1,46 +1,234 @@
-# Search spaces
-
-## General information
+# Search Spaces
 
 The term "search space" refers to the domain of possible values for the parameters that are being optimized during a campaign. A search space represents the space within which BayBE explores and searches for the optimal solution. It is implemented via the [`SearchSpace`](baybe.searchspace.core.SearchSpace) class.
 
-A search space might be purely discrete, purely continuous, or hybrid. This is encoded by the ``type`` attribute (see [`SearchSpaceType`](baybe.searchspace.core.SearchSpaceType)). Furthermore, it is possible to restrict certain types of search spaces using [`Constraints`](./constraints.md).
+BayBE offers three different kinds of search spaces, depending on the parameters that are used when defining a search space. A search space can be either __discrete__, __continuous__ or __hybrid__ which is reflected by the [`type`](baybe.searchspace.core.SearchSpaceType) property. BayBE offers different ways of constructing and working with the different kinds of search spaces.
+
+## Discrete search spaces
+
+A search space in which all parameters are discrete, i.e., in which all parameters are [`DiscreteParameter`](baybe.parameters.base.DiscreteParameter), is a **discrete search space**. BayBE offers different ways of constructing a discrete search space.
+
+### Building the full product space
+
+The method [`SearchSpace.from_product`](baybe.searchspace.core.SearchSpace.from_product) constructs the full cartesian product of the provided parameters:
+
+```python
+from baybe.parameters import NumericalDiscreteParameter
```

Comment:
For some weird reason, the code executability check fails on my machine due to some error regarding `NumericalDiscreteParameter`. Unfortunately, the error message is not helpful, and I am not able to reproduce it by extracting the code. In fact, even when copy-pasting the code that is created by the test and executing it manually, the error does not happen.
It would thus be great if @Scienfitz and @RimRihana could verify whether you get an error when running `pytest -s tests/docs/test_docs.py`.

Reply:
Looking at the (intermediate) CI tests, this also seems to fail during here. Will investigate, hopefully the error is easier to interpret here....

Reply:
Unfortunately not. Any idea what might cause this?!?
<img width="638" alt="image" src="https://github.com/emdgroup/baybe/assets/84016733/22a30061-7cab-4519-b5df-f18ac9425e1e">

Reply:
since it complains about the generator try to get rid of it and instead provide two or three hand constructed simple parameters. This would also open the opportunity to demonstrate eg a categorical parameter and a numerical one and their interplay in the resulting product

Reply:
fixed?

Reply:
At least locally for me. Will resolve once all tests pass.